segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -273,7 +273,7 @@ public CompletableFuture<SegmentHandle> openWrite(String streamSegmentName) {
                     .handleAsync( (v, ex) -> {
                         if (null != ex) {
                             log.debug("{} openWrite - exception segment={} latency={}.", logPrefix, streamSegmentName, timer.getElapsedMillis(), ex);
-                            handleException(streamSegmentName, ex);
+                            throw handleException(streamSegmentName, ex);
```

Comment:
This change is not needed.

The method doesn't actually return any value.
```
private RuntimeException handleException(String streamSegmentName, Throwable e) {
        val ex = Exceptions.unwrap(e);
        if (ex instanceof StorageMetadataWritesFencedOutException) {
            throw new CompletionException(new StorageNotPrimaryException(streamSegmentName, ex));
        }
        if (ex instanceof ChunkStorageFullException) {
            throw new CompletionException(new StorageFullException(streamSegmentName, ex));
        }
        throw new CompletionException(ex);
    }
```

Reply:
The method doesn't return any value it always throws. The reason the method signature shows it returning a RuntimeException is so that it can be called in this way. This does two things: 1) Makes it clear to someone looking at the call site that an exception is being thrown. 2) It helps with automated tools that make the assumption that methods will return when they are doing flow analysis. For example if a function ends in a call to handleException there will be an error about not calling return, when of course if there is a return call below it will be unreachable.